common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenSilentOperationParameters.java

```diff
@@ -61,8 +63,32 @@ public void validate() throws ArgumentException {
     }
 
     private boolean authorityMatchesAccountEnvironment() {
-        final AzureActiveDirectoryCloud cloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(mAccount.getEnvironment());
-        return cloud.getPreferredNetworkHostName().equals(getAuthority().getAuthorityURL().getAuthority());
+        final String methodName = ":authorityMatchesAccountEnvironment";
+        try {
+            if (!AzureActiveDirectory.isInitialized()) {
+                performCloudDiscovery();
+            }
+            final AzureActiveDirectoryCloud cloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(mAccount.getEnvironment());
+            return cloud != null && cloud.getPreferredNetworkHostName().equals(getAuthority().getAuthorityURL().getAuthority());
+        } catch (IOException e) {
+            Logger.error(
+                    TAG + methodName,
+                    "Unable to perform cloud discovery",
+                    e);
+            return false;
```

Comment:
What happens if this fails and we return false? Does the caller currently handle this in a reasonable way?

Reply:
We are just throw an ArgumentException in the case.
Since we are not sure if account matches the authority (as we don't have cloud information), we just say that the account does not match the authority and throw the exception.

Reply:
This is where this is called:

`    @Override
    public void validate() throws ArgumentException {
        super.validate();

        if (mAccount == null) {
            Logger.warn(TAG, "The account set on silent operation parameters is NULL.");
        } else if (!authorityMatchesAccountEnvironment()) {
            throw new ArgumentException(
                    ArgumentException.ACQUIRE_TOKEN_SILENT_OPERATION_NAME,
                    ArgumentException.AUTHORITY_ARGUMENT_NAME,
                    "Authority passed to silent parameters does not match with the cloud associated to the account."
            );
        }
    }`